fineract-provider/src/main/java/org/apache/fineract/infrastructure/core/config/SecurityConfig.java

```diff
@@ -80,6 +87,11 @@ protected void configure(HttpSecurity http) throws Exception {
                 .addFilterAfter(tenantAwareBasicAuthenticationFilter(), SecurityContextPersistenceFilter.class) //
                 .addFilterAfter(twoFactorAuthenticationFilter, BasicAuthenticationFilter.class); //
 
+        // Apply the HTTP method Filter only If the instance is Read or Write
+        if (fineractProperties.getMode().isReadOnlyMode() || fineractProperties.getMode().isWriteEnabled()) {
```

Comment:
Nah, let's apply the filter everytime. No need to exclude it when it's not running in in a certain mode.